tests/system/large/operations/test_semantics.py

```diff
@@ -130,6 +130,68 @@ def test_agg_invalid_cluster_column_raise_error(gemini_flash_model, cluster_colu
     df.semantics.agg(instruction, gemini_flash_model, cluster_column=cluster_column)
 
 
+@pytest.mark.parametrize(
+    ("n_clusters"),
+    [
+        pytest.param(1, id="one", marks=pytest.mark.xfail(raises=ValueError)),
+        pytest.param(2, id="two"),
+        pytest.param(4, id="four"),
+    ],
+)
+def test_cluster_by_w_n_clusters(session, text_embedding_generator, n_clusters):
```

Comment:
nit: just "test_cluster_by" should suffice. The varying numbers of clusters are already reflected in the function parameter

Reply:
Good to know. Thanks!